### PR TITLE
Remove ADO support/dependency from TableDefinition and ColumnDefinition. Fixes #704

### LINKED ISSUE #704: Remove ADO support/dependency from TableDefinition and ColumnDefinition
Since we need to touch ColumnDefinition to strip out encoding and codec, this sounds like the right time to do this.

### PR DESCRIPTION
**Pull request by jcferretti**

### CODE REVIEW

File: DB/src/main/java/io/deephaven/db/tables/TableDefinition.java [hunk 1]
```diff
@@ -742,34 +742,17 @@ public static TableDefinition createUserPartitionedTableDefinition(@NotNull fina
...
-            storageType = in.readInt();
-
-            // This read isn't using PersistentInputStream's ColumnSet conversion - need to consume stream elements for
-            // the columns I've removed.
-            in.readObject(); // Consume partitions
-            in.readUTF();    // Consume partionKey
-            in.readObject(); // Consume partitionerClass
-            in.readUTF();    // Consume partitionerArguments
-        }
+        namespace = readAdoString(in);
```
[the remark this answers is not in the document]
Comment (rcaudy) [reply DB/src/main/java/io/deephaven/db/tables/TableDefinition.java:745]: I understand if this is for a different PR.

File: DB/src/main/java/io/deephaven/db/tables/ColumnDefinition.java [hunk 2]
```diff
@@ -413,66 +410,49 @@ public boolean isFixedWidthObjectType() {
...
-            }
-
-            dataType = (Class)in.readObject();
-            componentType = (Class)in.readObject();
-            columnType = in.readInt();
-            isVarSizeString = (Boolean)in.readObject();
+        dataType = (Class)in.readObject();
+        componentType = (Class)in.readObject();
+        columnType = in.readInt();
+        isVarSizeString = (Boolean)in.readObject();
```
[the remark this answers is not in the document]
Comment (rcaudy) [reply DB/src/main/java/io/deephaven/db/tables/ColumnDefinition.java:440]: OK as later PR.

File: DB/src/main/java/io/deephaven/db/v2/NestedPartitionedDiskBackedTable.java [hunk 3]
```diff
@@ -34,7 +34,7 @@ public NestedPartitionedDiskBackedTable(TableDefinition tableDefinition,
                                             Set<String> internalPartitions) {
         //noinspection unchecked
         super(checkTableDefinitionRequirements(tableDefinition),
-                NestedPartitionedDiskBackedTable.class.getSimpleName() + '[' + tableDefinition.getNamespace() + ',' + tableDefinition.getName() + ']',
+                NestedPartitionedDiskBackedTable.class.getSimpleName(),
```
Comment (rcaudy) [inline DB/src/main/java/io/deephaven/db/v2/NestedPartitionedDiskBackedTable.java:37]: Would be better to replace the removed fragment with:
` + '[' + tableLocationProvider + ']'`

File: DB/src/main/java/io/deephaven/db/v2/SourceTable.java [hunk 4]
```diff
@@ -263,8 +263,6 @@ protected void instrumentedRefresh() {
                     //noinspection unchecked
                     return new EmptyToNullStringRegionedColumnSource(source, () ->
```
Comment (rcaudy) [inline DB/src/main/java/io/deephaven/db/v2/SourceTable.java:264]: You should just delete this method (`com.illumon.iris.db.v2.SourceTable#maybeReplaceColumnSources`), remove its single use, delete the field `com.illumon.iris.db.v2.SourceTable#MAP_EMPTY_STRINGS_TO_NULL`, and delete the class `com.illumon.iris.db.v2.sources.EmptyToNullStringRegionedColumnSource`.

File: DB/src/main/java/io/deephaven/db/v2/SourceTableMap.java [hunk 5]
```diff
@@ -73,7 +73,7 @@ public SourceTableMap(@NotNull final TableDefinition tableDefinition,
             subscriptionBuffer = new TableLocationSubscriptionBuffer(tableLocationProvider);
             pendingLocationStates = new IntrusiveDoublyLinkedQueue<>(IntrusiveDoublyLinkedNode.Adapter.<PendingLocationState>getInstance());
             readyLocationStates = new IntrusiveDoublyLinkedQueue<>(IntrusiveDoublyLinkedNode.Adapter.<PendingLocationState>getInstance());
-            processNewLocationsLiveTable = new InstrumentedLiveTable(SourceTableMap.class.getSimpleName() + '[' + tableDefinition.getNamespace() + ',' + tableDefinition.getName() + ']' + "-processPendingLocations") {
+            processNewLocationsLiveTable = new InstrumentedLiveTable(SourceTableMap.class.getSimpleName() + "-processPendingLocations") {
```
Comment (rcaudy) [inline DB/src/main/java/io/deephaven/db/v2/SourceTableMap.java:76]: A better choice would be to replace the removed code fragment with:
`'[' + tableLocationProvider + ']' +`

File: DB/src/main/java/io/deephaven/db/tables/ColumnDefinition.java [hunk 6]
```diff
@@ -259,14 +264,14 @@ private ColumnDefinition(String name, Class<TYPE> dataType, int columnType) {
     }
 
     private ColumnDefinition(String name, Class<TYPE> dataType, int columnType, boolean isVarSizeString) {
-        super(Objects.requireNonNull(name));
+        this.name = name;
```
Comment (rcaudy) [inline DB/src/main/java/io/deephaven/db/tables/ColumnDefinition.java:267]: You lost the `Objects.requireNonNull` in this change.

File: DB/src/main/java/io/deephaven/db/tables/ColumnDefinition.java [hunk 7]
```diff
@@ -480,118 +479,167 @@ public boolean isFixedWidthObjectType() {
 
     // TODO: DELETE THESE OVERRIDES (SEE NOTES IN TableDefinition)
```
Comment (rcaudy) [inline DB/src/main/java/io/deephaven/db/tables/ColumnDefinition.java:480]: You can and should delete this TODO now, as well as `MAGIC_NUMBER` (which has some crappy test use case that can be served some other way).

File: DB/src/main/java/io/deephaven/db/tables/ColumnDefinition.java [hunk 8]
```diff
@@ -480,118 +479,167 @@ public boolean isFixedWidthObjectType() {
...
 
-    private static final byte STRING_ENCODING_VERSION = 1;
-    private static final byte OBJECT_CODEC_VERSION = 2;
-    private static final byte CURRENT_VERSION = OBJECT_CODEC_VERSION;
-
     static final byte MAGIC_NUMBER = (byte)0b10001111;
 
-    public void readExternal(ObjectInput in) throws IOException, ClassNotFoundException {
-        ColumnsetConversionSchema conversionSchema = null;
+    protected String name;
```
Comment (rcaudy) [inline DB/src/main/java/io/deephaven/db/tables/ColumnDefinition.java:484]: These fields that came down from the superclass should be private.

File: DB/src/main/java/io/deephaven/db/tables/TableDefinition.java [hunk 9]
```diff
@@ -71,21 +61,11 @@ public TableDefinition(@NotNull final ColumnDefinition[] columnDefs) {
...
-        if (setColumns) {
-            this.setColumns(other.columns);
-            this.columnNameMap = other.columnNameMap;
-        }
+        this.setColumns(other.columns);
+        this.columnNameMap = other.columnNameMap;
     }
 
-    public static TableDefinition tableDefinition(@NotNull final Class<?>[] types, @NotNull final String[] columnNames) {
+    public static TableDefinition tableDefinition(@NotNull final Class[] types, @NotNull final String[] columnNames) {
```
Comment (rcaudy) [inline DB/src/main/java/io/deephaven/db/tables/TableDefinition.java:68]: Why did you drop the `<?>`, thus provoking a new raw types warning?

File: DB/src/main/java/io/deephaven/db/tables/TableDefinition.java [hunk 10]
```diff
@@ -592,60 +544,18 @@ public static TableDefinition createUserPartitionedTableDefinition(@NotNull fina
...
         columnDefs.addAll(baseDefs);
 
-        final TableDefinition result = new TableDefinition(columnDefs);
-        result.setStorageType(TableDefinition.STORAGETYPE_NESTEDPARTITIONEDONDISK);
-
-        return result;
+        return new TableDefinition(columnDefs);
     }
 
     // TODO: DELETE THESE OVERRIDES AND RELATED CODE IN persist.*() METHODS (SEE NOTES BELOW)
```
Comment (rcaudy) [inline DB/src/main/java/io/deephaven/db/tables/TableDefinition.java:550]: I think this TODO should be deleted.

File: DB/src/main/java/io/deephaven/db/tables/TableDefinition.java (hunk 9 above)
Comment (jcferretti) [reply DB/src/main/java/io/deephaven/db/tables/TableDefinition.java:68]: I went on a rampage adding <?> everywhere and then realized mid-way-through that it was going to not-end-well, so started "reverting".  This was likely a case of "reverted too much".

**Review by rcaudy — APPROVED** [review]

**Pull request merged**